dao/src/main/java/org/thingsboard/server/dao/DaoUtil.java

```diff
@@ -163,4 +164,17 @@ public static List<EntitySubtype> convertTenantEntityTypesToDto(UUID tenantId, E
         }
         return list;
     }
+
+    public static List<EntitySubtype> convertTenantEntityInfosToDto(UUID tenantId, EntityType entityType, List<EntityInfo> entityInfos) {
+        if (CollectionUtils.isEmpty(entityInfos)) {
+            return Collections.emptyList();
+        }
+
+        List<EntitySubtype> list = new ArrayList<>(entityInfos.size());
+        for (var info : entityInfos) {
+            list.add(new EntitySubtype(TenantId.fromUUID(tenantId), entityType, info.getName()));
```

Comment:
Extract ```TenantId.fromUUID(tenantId)``` to a separate variable outside the loop. Rewrite via Streams if you wish.